pkg/virt-handler/launcher-clients/launcher-clients.go

```diff
@@ -0,0 +1,326 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2025 The KubeVirt Authors.
+ *
+ */
+
+package launcher_clients
+
+import (
+	goerror "errors"
+	"fmt"
+	"io"
+	"net"
+	"path/filepath"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+
+	diskutils "kubevirt.io/kubevirt/pkg/ephemeral-disk-utils"
+	"kubevirt.io/kubevirt/pkg/safepath"
+	"kubevirt.io/kubevirt/pkg/util"
+	virtcache "kubevirt.io/kubevirt/pkg/virt-handler/cache"
+	cmdclient "kubevirt.io/kubevirt/pkg/virt-handler/cmd-client"
+	"kubevirt.io/kubevirt/pkg/virt-handler/isolation"
+)
+
+type LauncherClientsManager interface {
+	GetVerifiedLauncherClient(vmi *v1.VirtualMachineInstance) (client cmdclient.LauncherClient, err error)
+	GetLauncherClient(vmi *v1.VirtualMachineInstance) (cmdclient.LauncherClient, error)
+	GetLauncherClientInfo(vmi *v1.VirtualMachineInstance) *virtcache.LauncherClientInfo
+	CloseLauncherClient(vmi *v1.VirtualMachineInstance)
+	IsLauncherClientUnresponsive(vmi *v1.VirtualMachineInstance) (unresponsive bool, initialized bool, err error)
+}
+
+type launcherClientsManager struct {
+	virtShareDir         string
+	launcherClients      virtcache.LauncherClientInfoByVMI
+	podIsolationDetector isolation.PodIsolationDetector
+}
+
+func NewLauncherClientsManager(
+	virtShareDir string,
+	podIsolationDetector isolation.PodIsolationDetector,
+) LauncherClientsManager {
+
+	l := &launcherClientsManager{
+		virtShareDir:         virtShareDir,
+		launcherClients:      virtcache.LauncherClientInfoByVMI{},
+		podIsolationDetector: podIsolationDetector,
+	}
+
+	return l
+}
+
+func (l *launcherClientsManager) GetVerifiedLauncherClient(vmi *v1.VirtualMachineInstance) (client cmdclient.LauncherClient, err error) {
```

Comment:
Nit: We are mixing named return values with unnamed ones in other functions. it is very confusing to have both even in the same file. I would suggest using unnamed return so when reading the code it is clearer what is being returned.

Reply:
Agreed. However, this is just moved code and I'd rather keep it as is for now to limit the number of variables to investigate in case of failures... I suggest we defer this effort to later.